### PR TITLE
testing/ceph: upgrade to 11.1.1

### PR DESCRIPTION
**Pull request by dx9**

### CODE REVIEW

File: testing/ceph/APKBUILD [hunk 1]
```diff
@@ -114,59 +117,73 @@ _sysconfdir=/etc
 _udevrulesdir=/etc/udev/rules.d
 _python_sitelib=/usr/lib/python2.7/site-packages
 
-
-_builddir="$srcdir"/$pkgname-$pkgver
+_builddir=$srcdir/$pkgname-$pkgver
```
Comment (dx9) [inline testing/ceph/APKBUILD]: 😄  nope this if cruft, builddir works fine now. tks

File: testing/ceph/APKBUILD [hunk 2]
```diff
@@ -114,59 +117,69 @@ _sysconfdir=/etc
...
 package() {
-	cd "$_builddir"
-	cd build
-	make DESTDIR="$pkgdir" install
+	cd $builddir/build
+	make DESTDIR=$pkgdir install || return 1
 	cd ..
 
-	rm -f "$pkgdir"$_sysconfdir/init.d/ceph
+	rm -f $pkgdir$_sysconfdir/init.d/ceph
```
Comment (jirutka) [inline testing/ceph/APKBUILD]: Why have you removed quotes around pkgdir? Variables that contain externally provided paths (like `builddir`, `pkgdir`, `srcdir`, …) should be quoted.

File: testing/ceph/APKBUILD [hunk 3]
```diff
@@ -178,44 +191,18 @@ base() {
...
-
-	mkdir -p "$subpkgdir"$_sysconfdir/logrotate.d
-	mv "$pkgdir"$_sysconfdir/logrotate.d/ceph \
-		"$subpkgdir"$_sysconfdir/logrotate.d/ceph
-
-	mkdir -p "$subpkgdir"$_python_sitelib
-	mv "$pkgdir"$_python_sitelib/ceph_detect_init* "$subpkgdir"$_python_sitelib
-	mv "$pkgdir"$_python_sitelib/ceph_disk* "$subpkgdir"$_python_sitelib
-
+	_pkg $_docdir/ceph sample.ceph.conf sample.fetch_config
```
Comment (jirutka) [inline testing/ceph/APKBUILD]: Unfortunately APKBUILD does not run with `set -e` (yet, it’s on our TODO list to change it), so every statement that may fail, but should not fail, and is not the last statement of the function, should end with `|| return 1`.

File: testing/ceph/APKBUILD [hunk 4]
```diff
@@ -89,16 +88,20 @@ subpackages="
...
+	py-rgw:py_rgw
 	libradosstriper
 	libradosstriper-dev:libradosstriper_dev
 	py-rbd:py_rbd
 	py-cephfs:py_cephfs
 "
 
-if [ "$WITH_TESTS" = ON ]; then
+if [ -n "$CEPH_TEST_PKG" ]; then
 	subpackages="$subpackages $pkgname-test:ceph_test"
```
Comment (jirutka) [inline testing/ceph/APKBUILD]: Why is this conditional?

File: testing/ceph/APKBUILD [hunk 5]
```diff
@@ -114,59 +117,69 @@ _sysconfdir=/etc
...
-	   -DWITH_PYTHON3=OFF \
-	   -DWITH_TESTS=$WITH_TESTS \
-	   || return 1
-   make
+	export CEPH_BUILD_VIRTUALENV=$builddir
+
+	mkdir -p $builddir/build
+	cd $builddir/build
+
+	if [ -n "$MAKE_CHECK" ] || [ -n "$CEPH_TEST_PKG" ]; then
```
Comment (jirutka) [inline testing/ceph/APKBUILD]: Why is this conditional?

File: testing/ceph/APKBUILD (hunk 2 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: right, my bad

File: testing/ceph/APKBUILD (hunk 3 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: exit 1 is called in _pkg on failure. Is that ok?

File: testing/ceph/APKBUILD (hunk 4 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: So ya this is backwards. Test package should build by default.

File: testing/ceph/APKBUILD (hunk 5 above)
Comment (dx9) [inline testing/ceph/APKBUILD]: Ya this is not right. WITH_TESTS=off is faster, however that's irrelevant if test package is built by default. I'll ditch the "WITH_TEST" build optionality.

File: testing/ceph/APKBUILD (hunk 3 above)
Comment (jirutka) [inline testing/ceph/APKBUILD]: Definitely not, I overlooked that. Please never call `exit` in APKBUILD.

Comment (dx9) [inline testing/ceph/APKBUILD]: ok

**Comment by andypost** [comment]
latest release is 12.2.4 also depends on boost, ref #2960

**Pull request closed without merging**